package P3.Tugas;

public class Kontainer {
    private String nomorResi;
    private String namaPemilik;
    private double kapasitasMaksimal;
    private double beratMuatanSaatIni;

    public Kontainer(String nomorResi, String namaPemilik,
            double kapasitasMaksimal) {
        this.nomorResi = nomorResi;
        this.namaPemilik = namaPemilik;
        this.kapasitasMaksimal = kapasitasMaksimal;
        this.beratMuatanSaatIni = 0;
    }

    public String getNomorResi() {
        return nomorResi;
    }

    public String getNamaPemilik() {
        return namaPemilik;
    }

    public double getKapasitasMaksimal() {
        return kapasitasMaksimal;
    }

    public double getBeratMuatanSaatIni() {
        return beratMuatanSaatIni;
    }

    public void tambahMuatan(double berat) {
        if (berat <= 0) {
            System.out.println("Berat muatan harus lebih dari 0 kg.");
        } else if (beratMuatanSaatIni + berat > kapasitasMaksimal) {
            System.out.println("Maaf, berat muatan melebihi kapasitas maksimal kontainer!");
        } else {
            beratMuatanSaatIni += berat;
        }
    }

    public void turunkanMuatan(double berat) {
        double batasPembongkaran = beratMuatanSaatIni * 0.5;
        if (berat <= 0) {
            System.out.println("Berat yang diturunkan harus lebih dari 0 kg.");
        } else if (berat > batasPembongkaran) {
            System.out.println("Maaf, demi keselamatan, pembongkaran muatan satu kali jalan "+
                                "tidak boleh melebihi 50% dari muatan saat ini!"
            );
        } else {
            beratMuatanSaatIni -= berat;
        }
    }
}
